### PR TITLE
Explore: Change service in hook, un-skip test

### PR DESCRIPTION
**Pull request by gelicia**
**What is this feature?**

This is a follow up to #85878 which skipped a test that was intermittently failing remotely, and never failing locally. 

The error failed tests would report was (full context in the [drone logs](https://drone.grafana.net/grafana/grafana/171362/1/6))
```
FAIL public/app/features/explore/spec/queryHistory.test.tsx (38.335 s)
  ● Explore: Query History › adds recently added query if the query history panel is already open

    Expected test not to call console.error().

    If the error is expected, test for it explicitly by mocking it out using jest.spyOn(console, 'error').mockImplementation() and test that the warning occurs.

    Error: Uncaught [TypeError: Cannot read properties of undefined (reading 'get')]

      41 |         }
      42 |
    > 43 |         return getDatasourceSrv().get(pane.datasource);
         |                                  ^
      44 |       })
      45 |     )
      46 |       .then((results) => results.filter(isFulfilled).map((result) => result.value))
```

The source of that error was useExplorePageTitle.ts https://github.com/grafana/grafana/blob/main/public/app/features/explore/hooks/useExplorePageTitle.ts#L43

I am fairly certain that this is supposed to be `getDataSourceSrv()` from the runtime, as opposed to `getDatasourceSrv()` from plugins. I am also fairly certain that using the runtime will always succeed, as we set up the mock for that service when we set up these tests ([src](https://github.com/grafana/grafana/blob/main/public/app/features/explore/spec/helper/setup.tsx#L82)).

**Why do we need this feature?**

Fix intermittent failures.

### CODE REVIEW

File: public/app/features/explore/hooks/useExplorePageTitle.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export function useExplorePageTitle(params: ExploreQueryParams) {
           return Promise.reject();
         }
 
-        return getDatasourceSrv().get(pane.datasource);
+        return getDataSourceSrv().get(pane.datasource);
```
Comment (ifrost) [inline public/app/features/explore/hooks/useExplorePageTitle.ts:43]: It's better to import from runtime but I don't quite get why would it change anything. The one from plugins calls runtime anyway:

```typescript
import { getDataSourceSrv as getDataSourceService } from '@grafana/runtime';

export const getDatasourceSrv = (): DatasourceSrv => {
  return getDataSourceService() as DatasourceSrv;
};
```

Why would it work now 🤔 Is there a race condition that mocks the other import or clears the service?

**Comment by ifrost** [comment]
> (...) as we set up the mock for that service (...)

I think we mock BackendSrv, not DataSourceSrc (which is set directly https://github.com/grafana/grafana/blob/5305316f5abfee0e3f5990c3ee39ec4fed9e6b6b/public/app/features/explore/spec/helper/setup.tsx#L112)

**Comment by gelicia** [comment]
> I think we mock BackendSrv, not DataSourceSrc

My mistake. Still though, if we set dataSourceSrv for the tests, and we have a dependency using datasourceSrv that will-probably-eventually-but-not-aways get to dataSourceSrv, I could see why we have intermittent failures. It's really difficult for me to investigate this as I cannot replicate it locally. 

What are you suggesting I do?

**Comment by ifrost** [comment]
>  if we set dataSourceSrv for the tests, and we have a dependency using datasourceSrv that will-probably-eventually-but-not-aways get to dataSourceSrv, I could see why we have intermittent failures

> What else should I be considering?

Using runtime dependency makes way more sense so we need this change anyway. I don't understand how it would fix flakiness though :D 

However, since it's not possible to replicate locally, I would suggest re-running the build 10 times and merging the PR if it passes 10 times in a row. 

BTW. are you testing it locally while running all tests? Could it have something to do with running tests in parallel and some other tests cleanup 🤔

**Review by ifrost — APPROVED** [review]

**Comment by gelicia** [comment]
@ifrost really good points! Running it locally with all tests now, and have kicked off the first of 10 drone builds for testing.

**Comment by gelicia** [comment]
Remote runs
| Run  | Time | FE Test Status  | Link |
| ------------- | ------------- |------------- |------------- |
|1 | apr 11 7:57a |	passed | https://drone.grafana.net/grafana/grafana/171763/1/1 |
|2 | apr 11 8:14a |	passed | https://drone.grafana.net/grafana/grafana/171773/1/1 |
|3 | apr 11 9:28a |	passed | https://drone.grafana.net/grafana/grafana/171798/1/1 |
|4 | apr 11 9:53a |	passed | https://drone.grafana.net/grafana/grafana/171806/1/1 |
|5 | apr 11 10:28a | passed | https://drone.grafana.net/grafana/grafana/171825/1/1 |
|6 | apr 11 10:46a | passed | https://drone.grafana.net/grafana/grafana/171833/1/1 |
|7 | apr 11 11:02a | passed | https://drone.grafana.net/grafana/grafana/171841/1/1 |
|8 | apr 11 11:20a | passed | https://drone.grafana.net/grafana/grafana/171858/1/1 |
|9 | apr 11 12:55a | passed | https://drone.grafana.net/grafana/grafana/171906/1/1 |
|10 | apr 11 1:15p | passed | https://drone.grafana.net/grafana/grafana/171923/1/1 |

Locally, my tests had several failures but it didn't finish after over 3.5 hours and what was failing seemed to be unrelated. I will attempt to run it overnight but I think it is an issue with my machine trying to run the entire test suite.

**Pull request merged**